README.md
<br />
<div align="center">
  <h3 align="center">Likan</h3>

  <p align="center">
    一个牛逼的插件!
    <br />
    An awesome vscode extension!
  </p>
</div>

## Features

![product-screenshot](public/images/screen-shot-base.png)

<p align="center">process env</p>

![product-screenshot](public/images/screen-shot-path.png)

<p align="center">
没有tsconfig.json的别名映射跳转<br />
jump without tsconfig.json
</p>

![product-screenshot](public/images/screen-shot-linked_editing.png)

<p align="center">
链接标记(js, jsx, ts, tsx, vue) 和 创建文件时自动生成文件头文档注释<br />
linked editing(js, jsx, ts, tsx, vue) and doc comment
</p>

<div align="center">
  <img src="public/images/screen-shot-statusbar.png" />
</div>

<p align="center">
状态栏信息<br />
statusbar information
</p>

## Configuration

查询跳转时自动添加到后缀

File suffix automatically added when querying jumps.

- **default**:

```json
{
  "likan.language.exts": [".js", ".ts", ".jsx", ".tsx"]
}
```

查询跳转时别名映射

Query the alias mapping of jump files.

- **default**:

```json
{
  "likan.list.alias": {
    "@": "${root}/src",
    "~": "${root}"
  }
}
```

还有更多...

And more...

## License

Distributed under the MIT License. See `LICENSE` for more information.


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### [1.0.36](https://github.com/lirkang/likan/compare/v1.0.35...v1.0.36) (2022-09-24)

### [1.0.35](https://github.com/lirkang/likan/compare/v1.0.34...v1.0.35) (2022-09-24)

### [1.0.34](https://github.com/lirkang/likan/compare/v1.0.33...v1.0.34) (2022-09-24)

### [1.0.33](https://github.com/lirkang/likan/compare/v1.0.32...v1.0.33) (2022-09-24)

### [1.0.32](https://github.com/lirkang/likan/compare/v1.0.31...v1.0.32) (2022-09-24)


### Features

* 优化tag wrap, change case 修改为调用lodash的words函数, 转换模板字符串现在会找到第一个匹配的引号对 ([eed080d](https://github.com/lirkang/likan/commit/eed080d7374251bfadc8a86fa61be8d9f216d5a5))

### [1.0.31](https://github.com/lirkang/likan/compare/v1.0.30...v1.0.31) (2022-09-21)

### [1.0.30](https://github.com/lirkang/likan/compare/v1.0.29...v1.0.30) (2022-09-20)

### [1.0.29](https://github.com/lirkang/likan/compare/v1.0.28...v1.0.29) (2022-09-20)

### [1.0.28](https://github.com/lirkang/likan/compare/v1.0.27...v1.0.28) (2022-09-19)


### Features

* 添加keybinding ([dd3c6f6](https://github.com/lirkang/likan/commit/dd3c6f64527546883726d3af1107b2fd7ffd69e7))
* snippets优化 ([579c4e3](https://github.com/lirkang/likan/commit/579c4e343c4cafa96f2362ebb29a85f181ddf9bd))
* snippets优化 ([06a12b0](https://github.com/lirkang/likan/commit/06a12b0debe2f33ff6f33dac53d908e12c1b591c))

### [1.0.27](https://github.com/lirkang/likan/compare/v1.0.26...v1.0.27) (2022-09-18)


### Features

* 暂存 ([dc36a09](https://github.com/lirkang/likan/commit/dc36a09b6e46efa0e8744f25674119de3dab969d))

### [1.0.26](https://github.com/lirkang/likan/compare/v1.0.25...v1.0.26) (2022-09-17)

### [1.0.25](https://github.com/lirkang/likan/compare/v1.0.24...v1.0.25) (2022-09-16)

### [1.0.24](https://github.com/lirkang/likan/compare/v1.0.23...v1.0.24) (2022-09-16)

### [1.0.23](https://github.com/lirkang/likan/compare/v1.0.22...v1.0.23) (2022-09-16)

### [1.0.22](https://github.com/lirkang/likan/compare/v1.0.21...v1.0.22) (2022-09-16)

### [1.0.21](https://github.com/lirkang/likan/compare/v1.0.20...v1.0.21) (2022-09-16)


### Features

* 迁移到rollup ([bde94d3](https://github.com/lirkang/likan/commit/bde94d3f3ea4902911a22bbdb0dd67e01077a5de))
* 新增前后位置括号是否一致判断 ([11102ca](https://github.com/lirkang/likan/commit/11102ca105f1ec2daed02f4bb34cb82273ba0358))

### [1.0.20](https://github.com/lirkang/likan/compare/v1.0.19...v1.0.20) (2022-09-15)

### [1.0.19](https://github.com/lirkang/likan/compare/v1.0.18...v1.0.19) (2022-09-15)


### Features

* 移除node-fetch ([1d3392a](https://github.com/lirkang/likan/commit/1d3392af4406b7f3b8b4d03cf903ec9c5535f486))
* 移除open改为调用内置的api ([3468a45](https://github.com/lirkang/likan/commit/3468a45f90817e393dde06fe6b1c9f31057f140e))

### [1.0.18](https://github.com/lirkang/likan/compare/v1.0.17...v1.0.18) (2022-09-14)


### Features

* 修改打包进程, 引入dayjs和lodash-es, 启用并完善了转换字符串, 优化了filesize的显示时机 ([b98ffbc](https://github.com/lirkang/likan/commit/b98ffbc6624ac0806237be08c318a5ac094168d3))

### [1.0.17](https://github.com/lirkang/likan/compare/v1.0.16...v1.0.17) (2022-09-14)

### [1.0.16](https://github.com/lirkang/likan/compare/v1.0.15...v1.0.16) (2022-09-14)


### Features

* 暂时移除转化字符串, 获取gitignore模板时显示进度条, 命令面板显示切换为更精准的条件判断 ([1d64141](https://github.com/lirkang/likan/commit/1d64141680f257b95a0d915bb1b5a7246f69ffea))

### [1.0.15](https://github.com/lirkang/likan/compare/v1.0.14...v1.0.15) (2022-09-13)

### [1.0.14](https://github.com/lirkang/likan/compare/v1.0.13...v1.0.14) (2022-09-13)


### Features

* 新增添加到工作区 ([e4c361b](https://github.com/lirkang/likan/commit/e4c361bd44ca9d16c0ecee48427ff4bfc8a201eb))

### [1.0.13](https://github.com/lirkang/likan/compare/v1.0.12...v1.0.13) (2022-09-13)


### Bug Fixes

* bug fix ([502f254](https://github.com/lirkang/likan/commit/502f2545aa01e4aa2c92b20ae9403c6546711c65))

### [1.0.12](https://github.com/lirkang/likan/compare/v1.0.1...v1.0.12) (2022-09-12)

### [1.0.11](https://github.com/lirkang/likan/compare/v1.0.0...v1.0.11) (2022-09-12)


### Features

* 添加代码片段注释和视图菜单 ([25e3a3c](https://github.com/lirkang/likan/commit/25e3a3c33f1fcb14368cc6e086901ebbf3d22cf9))
* 添加代码片段注释和视图菜单 ([bc3ad8e](https://github.com/lirkang/likan/commit/bc3ad8ef94956629cac2e6c0482313140893ee4c))
* 引入normalize-path ([2b5a86d](https://github.com/lirkang/likan/commit/2b5a86d2df6f171e9a926a7832d62f3094d1b095))

### [1.0.1](https://github.com/lirkang/likan/compare/v1.0.0...v1.0.1) (2022-09-12)


### Features

* 添加代码片段注释和视图菜单 ([25e3a3c](https://github.com/lirkang/likan/commit/25e3a3c33f1fcb14368cc6e086901ebbf3d22cf9))
* 添加代码片段注释和视图菜单 ([bc3ad8e](https://github.com/lirkang/likan/commit/bc3ad8ef94956629cac2e6c0482313140893ee4c))
* 引入normalize-path ([2b5a86d](https://github.com/lirkang/likan/commit/2b5a86d2df6f171e9a926a7832d62f3094d1b095))

## [1.0.0](https://github.com/lirkang/likan/compare/v0.9.21...v1.0.0) (2022-09-12)


### Features

* 多项优化, 新增转化字符串 ([f23ec9f](https://github.com/lirkang/likan/commit/f23ec9f4ee54c03d5d32a1b6077978175be28058))
* test ([ad30375](https://github.com/lirkang/likan/commit/ad30375f2e4e7467c6b6207f5cb5f14c308ae40f))


### Bug Fixes

* 修复bug, 迁移到uri ([aa8df3e](https://github.com/lirkang/likan/commit/aa8df3e161ba5807dfaaa1007c2ef58c7d16359b))

### [0.9.21](https://github.com/lirkang/likan/compare/v0.9.20...v0.9.21) (2022-09-08)

### [0.9.20](https://github.com/lirkang/likan/compare/v0.9.19...v0.9.20) (2022-09-08)


### Features

* 新增更多配置项 ([b56a70f](https://github.com/lirkang/likan/commit/b56a70f57acb7193a3395ffa6268e294c6adef93))

### [0.9.19](https://github.com/lirkang/likan/compare/v0.9.18...v0.9.19) (2022-09-08)

### [0.9.18](https://github.com/lirkang/likan/compare/v0.9.17...v0.9.18) (2022-09-08)


### Features

* 移除图标, 对脚本列表进行排序, 现在change case 中数字将被分为同一组 ([e42bd61](https://github.com/lirkang/likan/commit/e42bd6167de20a56c9a79fa7ee752e3b6078425b))


rollup.config.js
// @ts-check

/**
 * @Author
 * @Date 2022-09-15 16:01:14
 * @Filepath likan/rollup.config.js
 * @Description
 */

import alias from '@rollup/plugin-alias';
import commonjs from '@rollup/plugin-commonjs';
import inject from '@rollup/plugin-inject';
import json from '@rollup/plugin-json';
import { nodeResolve } from '@rollup/plugin-node-resolve';
import replace from '@rollup/plugin-replace';
import typescript from '@rollup/plugin-typescript';
import { resolve } from 'path';
import { defineConfig } from 'rollup';
import filesize from 'rollup-plugin-filesize';
import { terser } from 'rollup-plugin-terser';

const IS_PROD = process.env.NODE_ENV !== 'development';

const config = defineConfig({
  input: 'src/index.ts',
  output: [{ format: 'commonjs', file: 'lib/index.js', sourcemap: IS_PROD ? false : 'inline' }],
  external: ['vscode'],
  treeshake: IS_PROD ? 'smallest' : false,
  watch: {
    include: ['src/**'],
    exclude: ['node_modules/**', 'lib/**'],
  },
  plugins: [
    inject({ vscode: 'vscode' }),
    typescript({ sourceMap: !IS_PROD }),
    nodeResolve({ extensions: ['.js', '.ts'], mainFields: ['module', 'main'] }),
    json({ preferConst: IS_PROD }),
    commonjs({}),
    replace({ preventAssignment: true }),
    alias({ entries: [{ find: '@', replacement: resolve(__dirname, 'src') }] }),
  ],
});

if (process.env.NODE_ENV !== 'development') {
  config.plugins?.push(filesize({}));
}

if (process.env.NODE_ENV === 'production') {
  config.plugins?.push(terser({ format: { comments: false }, compress: { drop_console: true }, mangle: true }));
}

export default config;
